src/stk500.c

```diff
@@ -727,11 +726,45 @@ static int stk500_parseextparms(const PROGRAMMER *pgm, const LISTID extparms)
       }
     }
 
+    else if (str_starts(extended_param, "xtal")) {
+      // Set clock generator frequency
+      if (str_starts(extended_param, "xtal=")) {
+        char xtal_str[16] = {0};
+        int sscanf_success = sscanf(extended_param, "xtal=%10s", xtal_str);
+        if (sscanf_success < 1) {
+          pmsg_error("invalid xtal value '%s'\n", extended_param);
+          rv = -1;
+          break;
+        }
+        char *endp;
+        double v = strtod(xtal_str, &endp);
+        if (endp == xtal_str){
```

Comment:
Would be neat to allow space between number and unit; eg, with code to the effect of
```
  if(*endp == ' ')
    endp++;
```

Reply:
Good POLA point, I will check.

Reply:
Users are unlikely to use quotes but scripts that extract the frequency from another AVRDUDE or otherwise call might have a frequency as `1.23 MHz` and find it easy to use quotes but marginally harder to remove the space between 3 and MHz. It is always neat if the output of AVRDUDE (correctly done with space between number and unit) can be used as input without mods.

Reply:
ok, c983e30 allows spaces in extra parameter, e.g. `-x "fosc= .1e3 kHz"`